### PR TITLE
Initial Prometheus Configuration

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: terraform_modules/prometheus_docker_resources/prometheus_ecr_images.tf [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+resource "docker_image" "bash" {
+  name = data.docker_registry_image.bash.name
+
+  pull_triggers = [data.docker_registry_image.bash.sha256_digest]
+}
+
+resource "null_resource" "tag_and_push_prometheus" {
+  triggers = {
```
Comment (bjpirt) [inline terraform_modules/prometheus_docker_resources/prometheus_ecr_images.tf:20]: Impressed by getting TF to pull and push to our own repository 👍

File: terraform_modules/prometheus_ecs/main.tf [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  tags = var.tags
+}
+
+resource "aws_lb_target_group" "prometheus_alb" {
+  name        = "${var.environment}-prometheus"
+  port        = 9090
+  protocol    = "HTTP" #tfsec:ignore:AWS004
+  vpc_id      = var.vpc_id
+  target_type = "ip"
+  slow_start  = 180
```
Comment (bjpirt) [inline terraform_modules/prometheus_ecs/main.tf:87]: Does it actually need slow start or was this copied from WAS?

File: terraform_modules/prometheus_ecs/main.tf [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  security_groups = [aws_security_group.prometheus_alb.id]
+  internal        = true
+
+  tags = var.tags
+}
+
+resource "aws_lb_target_group" "prometheus_alb" {
+  name        = "${var.environment}-prometheus"
+  port        = 9090
+  protocol    = "HTTP" #tfsec:ignore:AWS004
```
Comment (bjpirt) [inline terraform_modules/prometheus_ecs/main.tf:84]: Is there any way of it using HTTPS?

**Review by bjpirt — APPROVED** [review]

File: terraform_modules/prometheus_docker_resources/prometheus_ecr_images.tf (hunk 1 above)
Comment (brettminnie) [reply terraform_modules/prometheus_docker_resources/prometheus_ecr_images.tf:20]: Ta, was a nice little find

File: terraform_modules/prometheus_ecs/main.tf (hunk 2 above)
Comment (brettminnie) [reply terraform_modules/prometheus_ecs/main.tf]: No, that was in error, removed

File: terraform_modules/prometheus_ecs/main.tf (hunk 3 above)
Comment (brettminnie) [reply terraform_modules/prometheus_ecs/main.tf]: Yes there is, we'll need to create a self signed certificate and mount them in too, lets make a new ticket on that

**Review by bjpirt — APPROVED** [review]

**Pull request merged**